src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -385,6 +461,28 @@ void MarkCustomAttributes (ICustomAttributeProvider provider)
 			}
 		}
 
+		static AssemblyDefinition GetAssemblyFromCustomAttributeProvider (ICustomAttributeProvider provider)
```

Comment:
In my PR it came up that we'd all like to see less code living in the MarkStep.  One easy thing to do is to put simple static  utility type methods somewhere else.  I think this method could easily live else where.  An extension method would be nice and discoverable and would let you reduce the name down to `GetAssembly` 

i.e.
`static AssemblyDefinition GetAssembly (this ICustomAttributeProvider provider)`

Reply:
I agree but that should be done in separate PR, there are many more methods we could move from MarkStep